src/PockyBot.NET/Services/Triggers/RoleConfig.cs

```diff
@@ -0,0 +1,152 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GlobalX.ChatBots.Core;
+using GlobalX.ChatBots.Core.Messages;
+using PockyBot.NET.Constants;
+using PockyBot.NET.Persistence.Repositories;
+
+namespace PockyBot.NET.Services.Triggers
+{
+    internal class RoleConfig : ITrigger
+    {
+        private readonly IPockyUserRepository _pockyUserRepository;
+        private readonly IChatHelper _chatHelper;
+
+        public string Command => Commands.RoleConfig;
+
+        public bool DirectMessageAllowed => false;
+
+        public bool CanHaveArgs => true;
+
+        public string[] Permissions => new[] { Roles.Admin, Roles.Config };
+
+        public RoleConfig(IPockyUserRepository pockyUserRepository, IChatHelper chatHelper)
+        {
+            _pockyUserRepository = pockyUserRepository;
+            _chatHelper = chatHelper;
+        }
+
+        public async Task<Message> Respond(Message message)
+        {
+            var command = message.MessageParts[1].Text
+                .Trim()
+                .Remove(0, Command.Length)
+                .Trim()
+                .ToLowerInvariant();
+
+            if (string.IsNullOrWhiteSpace(command))
+            {
+                return new Message
+                {
+                    Text = "Please specify a command. Possible values are get, set, and delete."
+                };
+            }
+
+            string responseText;
+            switch (command)
+            {
+                case Actions.Get:
+                    responseText = await GetRoleConfigMessage().ConfigureAwait(false);
+                    break;
+                case Actions.Set:
+                    responseText = await SetRoleConfigMessage(message).ConfigureAwait(false);
+                    break;
+                case Actions.Delete:
+                    responseText = await DeleteRoleConfigMessage(message).ConfigureAwait(false);
+                    break;
+                default:
+                    responseText = "Unknown command. Possible values are get, set, and delete.";
+                    break;
+            }
+
+            return new Message
+            {
+                Text = responseText
+            };
+        }
+
+        private async Task<string> GetRoleConfigMessage()
+        {
+            var users = await _pockyUserRepository.GetAllUserRolesAsync();
+
+            if (users.Count == 0)
+            {
+                return "No roles have been set.";
+            }
+
+            var roleConfigMessageBuilder = new StringBuilder("Here is the current config:\n\n");
+            foreach (var user in users)
+            {
+                roleConfigMessageBuilder.Append(
+                    $"- {user.Username}: {string.Join(", ", user.Roles.Select(x => x.UserRole))}\n");
```

Comment:
Prefer AppendLine over Append with a manually-added newline, and adjust tests to ensure only testing in a system-agnostic way as much as possible.